frontend/lib/providers/test_results_search.dart

```diff
@@ -14,25 +14,24 @@
 // You should have received a copy of the GNU General Public License
 // along with this program.  If not, see <https://www.gnu.org/licenses/>.
 
-import 'package:dartx/dartx.dart';
 import 'package:riverpod_annotation/riverpod_annotation.dart';
 
 import '../models/detailed_test_results.dart';
-import '../models/execution_metadata.dart';
+import '../models/test_results_filters.dart';
 import 'api.dart';
 
 part 'test_results_search.g.dart';
 
 @riverpod
-class TestResultsSearchFromUri extends _$TestResultsSearchFromUri {
+class TestResultsSearch extends _$TestResultsSearch {
   @override
-  AsyncValue<TestResultsSearchResult> build(Uri pageUri) {
+  AsyncValue<TestResultsSearchResult> build(TestResultsFilters filters) {
```

Comment:
`testResultsSearch` now depends on `TestResultsFilters` instead of the page uri. This makes it super easy for other things to use this provider, as they just need to create the filters.